### DISCUSSION TITLE
on_user_update() does not show "before.avatar_url" when the avatar changes for the first time

### DISCUSSION THREAD

**Discussion by BrokenW1nd** [Rapptz/discord.py#6470] [Q&A]
"on_user_update ()" does not show "before.avatar_url" when the bot was first launched, or it was disabled and the user changed the avatar and then changed again when the bot was enabled.
I also noticed this problem in the popular "MEE6"
Can this be solved somehow? Or write in support?
Thanks

**Comment by esmaycat** [comment]
If the bot is not online and receiving events from discord (like the `on_user_update` event) then it wouldn't be able to detect a change that happened when it was not online and able to receive events. The `on_user_update` like any other event is only called when it actually happens.

I am not sure if this is the answer that you are looking for.

If it is for your own bot and it is online when you (or any other user) changes their avatar then it could also be a problem with your own intents, as your bot's client will only receive events like `on_user_update` if it has Intents.members` to be enabled like so:

```py
import discord
from discord.ext import commands

intents = discord.Intents.default()
intents.members = True
bot = commands.Bot(intents=intents) # The other parameters have been left out because they are unnecessary to explain
```

**Reply by BrokenW1nd** [reply]
Thx, but this has already been tested. This is a problem with discord.py and I solved this problem by creating a database that updates the avatars.
A bit more complicated than expected, but better this way =)
Thanks again for your reply.
